app/src/main/java/com/parasoft/demoapp/HomeActivity.java

```diff
@@ -99,17 +99,27 @@ public void loadOrderList(boolean loadFirstTime) {
                     public void onResponse(@NonNull Call<ResultResponse<OrderListResponse>> call,
                                            @NonNull Response<ResultResponse<OrderListResponse>> response) {
                         ordersLoadFinished();
-                        if (response.code() != 200) {
-                            showErrorView(getResources().getString(R.string.orders_loading_error));
-                            return;
-                        }
-                        List<OrderResponse> orderList = response.body() != null ? response.body().getData().getContent() : null;
-                        if (orderList == null || orderList.size() == 0) {
-                            showNoOrderView();
+                        if (response.code() == 200) {
+                            List<OrderResponse> orderList = response.body() != null ? response.body().getData().getContent() : null;
+                            if (orderList == null || orderList.size() == 0) {
+                                showNoOrderView();
+                            } else {
+                                showOrderListView(orderList);
+                            }
+                            orderItemClickable = true;
+                        } else if (response.code() == 400) {
```

Comment:
Maybe refactor line 110 to line 122 to conditionally set the errMsg and reduce call for methods.